Test/patterns/PatternMatching.dfy

```diff
@@ -0,0 +1,89 @@
+// RUN: %dafny /compile:3 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/*
+*  This file contains a collection of tests for features modified or introduced in PR 458
+*/
```

Comment:
Also add tests where the patterns use literals of various types: `bool`, `int`, some `newtype`, `real`, `char`, `string`. Are sequence display expressions (e.g., `[5, _, x]`) allowed? If so, test them, too. I'm guessing that display expressions for sets, multisets, and maps are not allowed (since their components are not ordered).